### PR TITLE
Update 1.3.10

### PR DESCRIPTION
**Pull request by Arishamays1**
## ☆Rasterio 1.3.1 Update ☆
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-4877)
[Upstream](https://github.com/mapbox/rasterio)
# Changes
- The Windows build was failing due to a GDAL-related problem upstream. Specifically, it was searching for `gdal_i_lib`, which does not exist in the GDAL libraries. Therefore, it was agreed to remove the reference to `gdal_i_lib` from `build.bat`.
- Added description
- Add skip for py<38
- Added new `dependencies`

### CODE REVIEW

**Review by JamesRobertsonGames — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -67,13 +69,15 @@ test:
     - $PYTHON -m pytest -v -m "not wheel" -rxXs -k "not (test_target_aligned_pixels or test_info_azure_unsigned)" tests  # [linux and build_platform == target_platform]
     - $PYTHON -m pytest -v -m "not wheel" -rxXs -k "not (test_target_aligned_pixels or test_outer_boundless_pixel_fidelity or test_info_azure_unsigned)" tests  # [linux and build_platform == target_platform]
     - $PYTHON -m pytest -v -m "not wheel" -rxXs -k "not (test_target_aligned_pixels or test_reproject_error_propagation or test_outer_boundless_pixel_fidelity)" tests  # [osx]
-    - python -m pytest -v -m "not wheel" -rxXs -k "not (test_hit_ovr or test_file_in_handler_with_vfs or test_files or test_parse_windows_path or test_copyfiles_same_dataset_another_name or test_context or test_target_aligned_pixels or test_decimated_no_use_overview or test_reproject_error_propagation)" tests  # [win]
+    # - python -m pytest -v -m "not wheel" -rxXs -k "not (test_decimated_no_use_overview or test_copyfiles_same_dataset_another_name)" tests  # [win]
```
Comment (danpetry) [inline recipe/meta.yaml]: should the win tests be commented out?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,17 +28,19 @@ requirements:
     - wheel
```
Comment (danpetry) [inline recipe/meta.yaml:28]: cython could be bounded, see [here](https://github.com/rasterio/rasterio/blob/560d4dd6b6c7fb402779ac815f83d618f357d013/pyproject.toml#L2)

File: recipe/meta.yaml [hunk 3]
```diff
@@ -28,17 +28,19 @@ requirements:
...
   run:
     - python
-
     - affine
     - attrs
+    - boto3 >=1.3.1
```
Comment (danpetry) [inline recipe/meta.yaml]: Just wondering where you got the boto3 and pyparsing pinnings from? The install_requires in setup.py define the runtime requirements, and they're defined [here](https://github.com/rasterio/rasterio/blob/560d4dd6b6c7fb402779ac815f83d618f357d013/setup.py#L264C1-L264C10) for this package

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: I plan on removing them altogether in my next commit to this PR as they are not necessary to exclude as these tests do not fail anymore for Windows. They are currently commented out to confirm that windows builds without any failing tests.

Comment (danpetry) [reply recipe/meta.yaml]: no tests are running at all, though - you commented out the entire `python -m pytest` line. is that what you wanted?

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: My mistake, boto3 is a required dep for testing, not run dependencies. I'll make an amendment to that. Pyparsing was a requirements that was also suggested for testing [here](https://github.com/rasterio/rasterio/blob/24d0845e576158217f6541c3c81b163d873a994d/requirements.txt#L11)

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: No, it isn't. Sorry about the oversight. I will revert the changes and restore the previous test configurations in my next commit to ensure the tests run as intended.

**Review by lorepirri — COMMENTED** [review]
@Arishamays1 if you are working on this version 

https://github.com/rasterio/rasterio/blob/1.3.10/setup.py

then the test requirements do not have `pyparsing`, which we do not have in the version that you've specified.

https://github.com/rasterio/rasterio/blob/560d4dd6b6c7fb402779ac815f83d618f357d013/setup.py#L283C1-L288C45


also, we do not use the `~` tilde notation (nor `^` caret) but we prefer to expand it (https://nodesource.com/blog/semver-tilde-and-caret/) to avoid confusion.

**Comment by Arishamays1** [comment]
> @Arishamays1 if you are working on this version
> 
> https://github.com/rasterio/rasterio/blob/1.3.10/setup.py
> 
> then the test requirements do not have `pyparsing`, which we do not have in the version that you've specified.
> 
> https://github.com/rasterio/rasterio/blob/560d4dd6b6c7fb402779ac815f83d618f357d013/setup.py#L283C1-L288C45
> 
> also, we do not use the `~` tilde notation (nor `^` caret) but we prefer to expand it (https://nodesource.com/blog/semver-tilde-and-caret/) to avoid confusion.

Thank you for the clarification, Lorenzo! I think I might've been looking in the wrong tag regarding the pyparsing dependecy.

**Review by danpetry — APPROVED** [review]

**Pull request merged**